tests/unit/authenticators/oauth2-password-grant-test.js

```diff
@@ -228,6 +238,38 @@ describe('OAuth2PasswordGrantAuthenticator', () => {
         });
       });
     });
+
+    describe('when the authentication request fails without a valid response', function() {
+      beforeEach(function() {
+        server.post('/token', () => [500, { 'Content-Type': 'text/plain', 'X-Custom-Context': 'foobar' }, 'The server has failed completely.']);
+      });
+
+      it('rejects with the response body', function(done) {
+        authenticator.authenticate('username', 'password').catch((error) => {
+          expect(error).to.eql('The server has failed completely.');
+          done();
+        });
+      });
+
+      describe('when rejectWithResponse is enabled', function() {
+        beforeEach(function() {
+          authenticator.set('rejectWithResponse', true);
+        });
+
+        it('rejects with xhr object containing responseText', function() {
```

Comment:
I'd say _"response object"_ here instead. With `fetch` this will not be an xhr anymore.

Reply:
Quite right.